apiary.apib

```diff
@@ -248,11 +248,75 @@ You can find a (very) minimal PHP code example, showing how start the OAuth 2 au
 
 A Teamleader PHP SDK is under development. In the meanwhile, we recommend using packages such [Guzzle](http://docs.guzzlephp.org/en/stable) or the [League OAuth2 client](http://oauth2-client.thephpleague.com).
 
-## Changelog
+## Changes & upgrades
 
-You can follow our API progress and ongoing proposals via our [API Blueprint GitHub repository](https://github.com/teamleadercrm/api).
+We want to keep this API backwards compatible as long as possible, but at the same time we need to make changes to it to match our evolving product.
+
+Therefor we assigned a dated version to your API client when it was created.
+This version controls the API and webhooks behaviour for you API client.
+When we change the API in a backwards-incompatible way, we release a new dated version.
+But to avoid breaking your code, you stay on your assigned version until you [upgrade your API version](./#upgrading-your-api-version).
+
+### Types of changes
+
+We consider the following changes to be backwards-compatible:
+
+- Adding new API endpoints.
+- Adding new optional parameters to existing API requests.
+- Adding new properties to existing API responses.
+- Adding new webhook types.
+- Making existing parameters optional.
+- Changing the order of properties in existing API responses.
+- Fixing (potential) bugs that made an endpoint unusable (HTTP 500 errors).
+- (Currently we also hold ourselves the right to return a different error format, as there is still room for improvement)
+
+We consider the following changes to be backwards-incompatible:
+
+- Renaming or removing an endpoint.
+- Renaming or removing a parameter of API requests.
+- Renaming or removing a property of API responses.
+- Changing the format of a parameter of API responses (eg. integer to float)
+- Making validation of a parameter more strict (eg. making it required)
+- Returning a different HTTP status code.
+
+### Upgrading your API version
+
+This documentation only reflects the latest version of the API.
+If you need to make changes to your client or you want to make use of the latest feature, you might need to upgrade your API version.
+
+You can find your current API version, in the `X-Api-Version` header on any API response.
+The current latest version is **2017-12-06**.
+
+After checking the API changelog to see which endpoints work differently,
+you can upgrade your API version in two ways:
+
+- Sending a `X-Api-Version` header with the request, with a newer version than your current version
```

Comment:
wondering how you should actually do that... you write X, so I assumed `a`... but you say "ix", so might need to be an `an`

Reply:
It's definitely `an`.
It's always based on the sound rather than the actual letter.
e.g. the Y is a vowel, but you treat it as a consonant since it's pronounced `why` -> `busting a Y.M.C.A. dance move` instead of `an Y.M.C.A. dance move`

Reply:
And fyi it is pronounced `eks` not `ix` :P